src/riscv/lib/src/machine_state/memory/state.rs

```diff
@@ -17,6 +21,14 @@ use crate::state_backend::ManagerRead;
 use crate::state_backend::ManagerReadWrite;
 use crate::state_backend::ManagerWrite;
 
+fn get_page_range(address: Address, length: usize) -> Range<usize> {
```

Comment:
nit, using `RangeInclusive` would save the `saturating_add(1)`, which isn't needed below